### PR TITLE
Move logic deciding whether to run E2E to its own script

### PR DESCRIPTION
**Pull request by mem**
This makes it easier to test and update the code.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,42 +64,10 @@ x-run:
...
-              ;;
-          *)
-              # The branch is not master or release, skip e2e
-              require_e2e=false
-              ;;
-      esac
-
-      if ! "$require_e2e" ; then
-        echo "No changes in critical components, skipping tests"
+      if scripts/should-e2e-run ; then
```
Comment (dtrudg) [inline .circleci/config.yml]: We need to pass the user/repo/pr here, right?

```
E: Invalid usage. Abort.

Usage:

    scripts/should-e2e-run {github_username} {github_reponame} {pr_number}
+ echo 'No critical changes, skipping tests'
No critical changes, skipping tests
+ circleci step halt
```

**Review by dtrudg — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (dtrudg) [reply .circleci/config.yml]: `scripts/should-e2e-run $CIRCLE_PR_USERNAME $CIRCLE_PR_REPONAME $CIRCLE_PR_NUMBER` ?

File: scripts/should-e2e-run [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
```
Comment (cclerget) [inline scripts/should-e2e-run:1]: Could this be moved to `.circleci` directory instead to not mix CI and build related scripts ?

File: .circleci/config.yml (hunk 1 above)
Comment (mem) [reply .circleci/config.yml]: Yeah, thanks for catching that. I could have sworn I had that...

File: scripts/should-e2e-run (hunk 2 above)
Comment (mem) [reply scripts/should-e2e-run:1]: No, I'd prefer to have things in the same visible location than reproduce the mess that is the Travis setup.

Comment (cclerget) [reply scripts/should-e2e-run:1]: And reproduce the same mess in `scripts` :/ ?

Comment (mem) [reply scripts/should-e2e-run:1]: I'll just assume you haven't looked at how travis is set up and that's why my comment is not understood.

Thanks!

File: .circleci/config.yml [hunk 3]
```diff
@@ -64,45 +64,24 @@ x-run:
   check_changes: &check_changes
     name: Check changes
     command: |-
-      # Skip e2e by default
-      require_e2e=false
+      rc=0
+      scripts/should-e2e-run "${CIRCLE_PROJECT_USERNAME}" "${CIRCLE_PROJECT_REPONAME}" "${CIRCLE_PULL_REQUEST}" || rc=$?
```
Comment (mem) [inline .circleci/config.yml:68]: For some reason the other variable is empty. Looking at the environment where _this_ check is running, that variable is not available. Only CIRCLE_PULL_REQUEST is. So use that and extract the PR number that we need from there.

Comment (dtrudg) [reply .circleci/config.yml:68]: heh - I just got it from their docs page somewhere. fair enough!

**Review by dtrudg — APPROVED** [review]

**Pull request merged**